File: Java/DesignPatterns/src/main/java/Facade/CPU.java
package Facade;

class CPU {
    private boolean isInitialized = false;

    public void init() {
        System.out.println("CPU is initialized");
        isInitialized = true;
    }

    public void processData(byte[] data) {
        if (!isInitialized) {
            return;
        }

        System.out.println("Processing data:");
        for (byte b : data) {
            System.out.printf("%02X%n", b);
        }
    }
}

File: C#/DesignPatterns/Facade/CPU.cs
namespace Facade
{
	public class CPU
	{
		bool isInitialzied = false;
		public void Init()
		{
			Console.WriteLine("CPU is initialized");
			isInitialzied = true;
		}

		public void ProcessData(byte[] data)
		{
			if (!isInitialzied)
			{
				return;
			}

			Console.WriteLine("Processing data:");

			foreach (var b in data)
			{
				Console.WriteLine(b.ToString("X2"));
			}
		}
	}
}